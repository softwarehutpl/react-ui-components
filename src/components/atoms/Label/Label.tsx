import React from 'react';
import styled, { css } from 'styled-components';

export interface LabelProps {
  children?: any;
  className?: string;
  disabled?: boolean;
};

const Label: React.StatelessComponent<LabelProps> = ({ 
  children, 
  className ,
}) => (
  <label className={className}>{children}</label>
);

export default styled(Label)`
  display: flex;
  cursor: pointer;
  position: relative;
  align-items: center;
  ${({ disabled }) =>
    disabled && 
      css`
        cursor: not-allowed;
        opacity: 0.4;
    `
  }
`;
