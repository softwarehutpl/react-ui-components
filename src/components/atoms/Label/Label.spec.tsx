import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Label from './Label';

describe('<Label />', () => {
  it('should render corectly', () => {
    const tree = renderer
    .create(<Label />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have properly style if disabled', () => {
    const tree = renderer
    .create(<Label disabled/>)
    .toJSON();
    expect(tree).toHaveStyleRule('opacity', '0.4');
    expect(tree).toHaveStyleRule('cursor', 'not-allowed');
  });
});
