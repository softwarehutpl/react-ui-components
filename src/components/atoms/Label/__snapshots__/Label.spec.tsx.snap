// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should render corectly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<label
  className="c0"
/>
`;
